scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/ScalametaTokenizer.scala

```diff
@@ -277,10 +277,7 @@ class ScalametaTokenizer(input: Input, dialect: Dialect)(implicit options: Token
 
       case EOF => new Token.EOF(input, dialect)
       case SHEBANG => new Token.Shebang(input, dialect, curr.offset, curr.endOffset, curr.strVal)
-      case INVALID => getInvalid(curr, curr.strVal)
-
-      // the rest of tokens shouldn't be obtained via this method
-      case _ => unreachable(debug(curr))
+      case _ => getInvalid(curr, curr.strVal)
```

Comment:
so what type of token was attempted here? just want to be sure we're not masking a problem elsewhere.so what type of token was attempted here? just want to be sure we're not masking a problem elsewhere.

Reply:
STRINGPART since we have invalid token on single $ we then have a normal string contents

Reply:
keeping `getInvalid` is a good idea, just in case. do you think we should adjust `getInvalid` to include the `LegacyToken` that was parsed? for debugging purposes?

Reply:
I split the cases again and added a proper message which includes that info whenever unexpected token was found